conans/client/source.py

```diff
@@ -192,20 +193,32 @@ def _run_scm(conanfile, src_folder, local_sources_path, output, cache):
     dest_dir = os.path.normpath(os.path.join(src_folder, scm_data.subfolder))
     if cache:
         # When in cache, capturing the sources from user space is done only if exists
-        captured = local_sources_path and os.path.exists(local_sources_path)
+        if not local_sources_path or not os.path.exists(local_sources_path):
+            local_sources_path = None
     else:
         # In user space, if revision="auto", then copy
-        captured = scm_data.capture_origin or scm_data.capture_revision
-    local_sources_path = local_sources_path if captured and conanfile.develop else None
-
-    if local_sources_path:
+        if scm_data.capture_origin or scm_data.capture_revision:  # FIXME: or clause?
+            scm = SCM(scm_data, local_sources_path, output)
+            scm_url = scm_data.url if scm_data.url != "auto" else \
+                scm.get_qualified_remote_url(remove_credentials=True)
+
+            if scm_url:  # If there is no remote (git), we cannot compute this block
+                src_path = scm.get_repo_root()
```

Comment:
The same difficult to understand block than above. Reinforce my idea that this might be abstracted away in SCM components

Reply:
Moved to a function inside SCM, still executed for every CVS